component/src/main/java/org/exoplatform/rhmanagement/services/rest/RequestRestService.java

```diff
@@ -0,0 +1,307 @@
+package org.exoplatform.rhmanagement.services.rest;
+
+
+import org.exoplatform.rhmanagement.dto.OfficialVacationDTO;
+import org.exoplatform.rhmanagement.dto.UserRHDataDTO;
+import org.exoplatform.rhmanagement.dto.VacationRequestDTO;
+import org.exoplatform.rhmanagement.dto.ValidatorDTO;
+import org.exoplatform.rhmanagement.services.*;
+import org.exoplatform.services.log.ExoLogger;
+import org.exoplatform.services.log.Log;
+import org.exoplatform.services.rest.resource.ResourceContainer;
+import org.exoplatform.services.organization.*;
+import org.exoplatform.services.security.ConversationState;
+import org.exoplatform.social.core.identity.model.Profile;
+import org.exoplatform.social.core.identity.provider.OrganizationIdentityProvider;
+import org.exoplatform.social.core.manager.IdentityManager;
+import org.exoplatform.social.core.space.model.Space;
+import org.exoplatform.social.core.space.spi.SpaceService;
+import org.exoplatform.social.service.rest.RestChecker;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import javax.annotation.security.RolesAllowed;
+import javax.servlet.http.HttpServletRequest;
+import javax.ws.rs.*;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.UriInfo;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Created by Medamine on 05/01/2017.
+ */
+
+@Path("/rhrequest")
+@Produces("application/json")
+public class RequestRestService implements ResourceContainer {
+
```

Comment:
Some info logs should be added to the end points methods. They will allow to detect potential intrusions when happened